assembly/utils.ts

```diff
@@ -745,33 +773,49 @@ function addTime(hour :i32, minute :i32, second :i32, millisecond :i32, microsec
   return balanceTime(hour, minute, second, millisecond, microsecond, nanosecond);
 }
 
-function balanceTime(hour :i32, minute :i32, second :i32, millisecond :i32, 
-  microsecond: i32, nanosecond: i32): BalancedTime {
-  
-  microsecond += floorDiv(nanosecond, 1000);
-  nanosecond = nonNegativeModulo(nanosecond, 1000);
-  millisecond += floorDiv(microsecond, 1000);
-  microsecond = nonNegativeModulo(microsecond, 1000);
-  second += floorDiv(millisecond, 1000);
-  millisecond = nonNegativeModulo(millisecond, 1000);
-  minute += floorDiv(second, 60);
-  second = nonNegativeModulo(second, 60);
-  hour += floorDiv(minute, 60);
-  minute = nonNegativeModulo(minute, 60);
-  var deltaDays = floorDiv(hour, 24);
-  hour = nonNegativeModulo(hour, 24);
```

Comment:
In this case we could avoid `nonNegativeModulo` util due to we already uses `floorDiv`.
`nonNegativeModulo(a, b)` -> `a - b * floorDiv(a, b)`